src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -1921,34 +1967,66 @@ private void doGenKeyPair(String alias, String dname, String keyAlgName,
             keypair.generate(keysize);
         }
 
-        PrivateKey privKey = keypair.getPrivateKey();
-
-        CertificateExtensions ext = createV3Extensions(
-                null,
-                null,
-                v3ext,
-                keypair.getPublicKeyAnyway(),
-                null);
+        CertificateExtensions ext;
+        if (signerAlias != null) {
+            ext = createV3Extensions(
+                    null,
+                    null,
+                    v3ext,
+                    keypair.getPublicKeyAnyway(),
+                    signerSubjectKeyId);
+        } else {
```

Comment:
No need for an if-else block? `signerSubjectKeyId` is null in this case.

Reply:
Removed.